api/capi/include/nntrainer_internal.h

```diff
@@ -313,6 +313,48 @@ int ml_train_model_insert_layer(ml_train_model_h model, ml_train_layer_h layer,
                                 const char *input_layer_names[],
                                 const char *output_layer_names[]);
 
+/**
+ * @brief Compiles and finalizes the neural network model with the params.
+ * @details Use this function to initialize neural network model. Various
+ * hyperparameter before compile the model can be set. Once compiled,
+ * any modification to the properties of model or layers/dataset/optimizer in
+ * the model will be restricted. Further, addition of layers or changing the
+ * optimizer/dataset of the model will not be permitted.
+ * @param[in] model The NNTrainer model handle.
+ * @param[in] singgle_param hyperparameters for compiling the model
+ * @return @c 0 on success. Otherwise a negative error value.
+ * @retval #ML_ERROR_NONE Successful.
+ * @retval #ML_ERROR_NOT_SUPPORTED Not supported.
+ * @retval #ML_ERROR_INVALID_PARAMETER Invalid parameter.
+ * API to solve va_list issue of Dllimport of C# interop.
+ * The input format of singgle_param muse be 'key = value' format, and it
+ * received as shown in the example below. delimiter is '`'. e.g)
+ * ml_train_model_compile_with_singgle_param(model, "loss = cross`batch_size =
+ * 9")
+ */
+int ml_train_model_compile_with_singgle_param(ml_train_model_h model,
```

Comment:
typo: singgle
Please fix this typo. This typo is everywhere

Reply:
yes, copy and paste is wrong. thank you.